### ISSUE TITLE
Make Comms Data accessible to parent

### ISSUE THREAD

**Issue by ttocsneb** [ttocsneb/rcProtocol#5]
This is a bit of a weird issue, but I need to have comms data more easily accessible, and writeable.  At the moment, everything works but is not developer friendly.  

To resolve this I will restructure the update function to not have any parameters.  There will be functions to get pointers the sending/receiving data.  Since this is a pointer function, you should only need to call it when a connection is made (You would need to reset it every time after a connection because the arrays may change size, and therefore the pointer).

**Comment by ttocsneb** [comment]
The last two channels are set to 0 or 1, the received channels vary between the actual values and max values.  I don't know why this is.  If it is because of the way memcpy works, or something else

**Comment by ttocsneb** [comment]
It also seems that update() will always return 22 (RC_INFO_TICK_TOO_SHORT)

**Comment by ttocsneb** [comment]
It turns out that I was not resetting the timer.

Now, looking at the telemetry packet, the entire packet seems to be shifted over one byte every once and a while.    There is no pattern that I can see, but it shifts to the right, whatever was at the end is now at the beginning.

This is happening about 50% of the time.

**Comment by ttocsneb** [comment]
I'm thinking that memcpy() is the culprit, so I will try rewriting my own version, or completely remove the usage of it.

**Comment by ttocsneb** [comment]
That wasn't it, It seems that the data the receiver gets is also a mess.  I have no idea what could be causing this, maybe when I changed it to use dynamic pointers.

I'll look into that.

**Comment by ttocsneb** [comment]
After reverting back to 56a1e57ef1f280b6ab30b75f3060a0502bf3f882 the first 4 channels always work, however, the 5 channel varies between being correct and some random number.  The 6 channel is always wrong.

**Comment by ttocsneb** [comment]
After removing the extra bits used to determine the type of packet, none of the contents of a sending packet were messed up.  This ends up removing every instance of memcpy.  I might still be focused too much on that being the problem, but I think that memcpy, and possibly my rc_cpy were both causing issues. That or the way I was using it was the issue.

However, when it comes to telemetry,  everything is received ok, but every once and a while the message is shifted one byte to the right.

**Comment by ttocsneb** [comment]
I was able to fix it.  I'm still not completely sure why, but in DeviceProtocol, I moved writeAckPayload into _checkPacket().  _radio.writeAckPayload was changed to use the pipe from _radio.available, I don't think this changed anything though.

While looking through the code I found that at [rcDeviceProtocol.cpp:233](https://github.com/ttocsneb/rcProtocol/blob/60d121cf14f54fce6853f21328da77acb97fc2d2/src/rcDeviceProtocol.cpp#L257) it loops while _checkPacket is 0.  It should be looping on 1 because 1 is returned when a packet is received.

I think that because the loop was active while _checkPacket didn't receive anything. The buffer for _radio.writeAckPayload() was getting overloaded, causing the received ack payloads to get messed up.
